components/board/i2c.go

```diff
@@ -54,3 +56,23 @@ func (reg *I2CRegister) ReadWordData(ctx context.Context) (uint16, error) {
 func (reg *I2CRegister) WriteWordData(ctx context.Context, data uint16) error {
 	return reg.Handle.WriteWordData(ctx, reg.Register, data)
 }
+
+// I2CAttrConfig is used for converting config attributes necessary
+// for establishing an I2C connection.
+type I2CAttrConfig struct {
+	I2CBus      string `json:"i2c_bus"`
+	I2cAddr     int    `json:"i2c_addr,omitempty"`
+	I2CBaudRate int    `json:"i2c_baud_rate,omitempty"`
+}
+
+// ValidateI2C ensures all parts of the config are valid.
+func (cfg *I2CAttrConfig) ValidateI2C(path string, requireAddress bool) error {
```

Comment:
An idea, but ugly: to make the change less breaking, could we pass in the entire config? Then check for the type (if it is an i2c config)? and use the functions below.

If it is not, could we check the fields for the flatter level config parameters, as in ``config.I2CAddr = ""``. Then validate using those attributes?

Otherwise, this breaks a lot of little sensors and motors everywhere, but I don't know exactly who is using i2c on a robot right now.